src/user/picture.js

```diff
@@ -90,7 +90,11 @@ module.exports = function(User) {
 	};
 
 	User.uploadFromUrl = function(uid, url, callback) {
-		var filename = 'uid:' + uid + ':tmp-image';
+		var extension = url.substring(url.lastIndexOf('.') + 1);
+		if (['png', 'jpeg', 'jpg', 'gif'].indexOf(extension) == -1) {
+			return callback('This image type is not allowed');
```

Comment:
Changed to error:invalid-image-extension (i think it is ok since image types are not configurable by user)